### PR TITLE
[Gutenberg] Use tarball instead of git tag for forked dependencies

### PR DESCRIPTION
**Pull request by ceyhun**
Gutenberg Mobile PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/3982

This PR only bumps the Gutenberg Mobile reference.

**To test:**
Follow testing instructions from https://github.com/WordPress/gutenberg/pull/34886.

## Regression Notes
1. Potential unintended areas of impact
None apart from the editor.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Smoke testing that the editor could be opened and blocks could be added.

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: Podfile.lock [hunk 1]
```diff
@@ -62,7 +62,7 @@ PODS:
     - AppAuth/Core (~> 1.4)
     - GTMSessionFetcher/Core (~> 1.5)
   - GTMSessionFetcher/Core (1.6.1)
-  - Gutenberg (1.62.0):
+  - Gutenberg (1.61.1):
```
Comment (hypest) [inline Podfile.lock]: WDYT about updating from develop(s) @fluiddot ? Looks like the tag here is older than the one on develop.

Comment (fluiddot) [reply Podfile.lock]: Sure, I'll update it, besides, there's a file conflict that should be resolved 👍 .

Comment (fluiddot) [reply Podfile.lock]: I updated the branch with `develop` in [this commit](https://github.com/wordpress-mobile/WordPress-iOS/pull/17198/commits/3852874a673a6496cd48fde8b7c8ecc47276d30a).

**Review by hypest — APPROVED** [review]
LGTM!

Before merging the main apps PRs, we shall create an alpha tag to point to the gutenberg-mobile merged version, right?

**Comment by fluiddot** [comment]
> LGTM!
> 
> Before merging the main apps PRs, we shall create an alpha tag to point to the gutenberg-mobile merged version, right?

Exactly, we'll create an alpha tag after merging the gutenberg-mobile PR and update the reference here.

**Comment by fluiddot** [comment]
> > LGTM!
> > Before merging the main apps PRs, we shall create an alpha tag to point to the gutenberg-mobile merged version, right?
> 
> Exactly, we'll create an alpha tag after merging the gutenberg-mobile PR and update the reference here.

The Gutenberg Mobile reference now points to the tag `v1.63.0-alpha1`.

**Pull request merged**